lib/src/default_index/composite.rs

```diff
@@ -543,11 +537,6 @@ impl<I: AsCompositeIndex + Send + Sync> ChangeIdIndex for ChangeIdIndexImpl<I> {
         }
     }
 
-    /// Calculates the shortest prefix length of the given `change_id` among all
-    /// ids including hidden entries.
-    ///
-    /// The returned length is usually a few digits longer than the minimum
-    /// length to disambiguate within the visible entries.
```

Comment:
removed by mistake? It says that the default index implementation doesn't return the minimum shortest length, though the implementation still conforms to the requirement in a way that the returned prefix can disambiguate a commit.

Reply:
ditto. change both paragraphs into non-doc comment, then?

The second paragraph doesn't explain why length is a few digits longer.

Reply:
I think I don't really understand this part:

```
    /// The returned length is usually a few digits longer than the minimum
    /// length to disambiguate within the visible entries.
```

If we're calculating the shortest prefix length to disambiguate among all entries (including hidden ones), then wouldn't that naturally disambiguate amongst visible entries? Isn't the set of visible entries a strict subset of the set of all entries? Why do we need to extend the length to disambiguate?

Reply:
I suspect it's a matter of parsing the sentence. I'm guessing you parsed it like this:

"The returned length is usually {a few digits longer than the minimum length} [in order] to disambiguate within the visible entries."

But I think Yuya meant this:

"The returned length is usually a few digits longer than {the minimum length to disambiguate within the visible entries}."

Reply:
Ah, yeah that would explain it. We could just add the word "necessary" to make this clearer:

"The returned length is usually a few digits longer than the minimum length **necessary** to disambiguate within the visible entries."

But do we even need this comment? I think deleting this from composite index is fine since the documentation in the trait declaration seems clear to me. If this is an important point to emphasize I can move it to the trait though.

Thoughts?

Reply:
Adding "necessary" sounds good. I think it's useful to have it there to explain how this implementation behaves. Otherwise a reader might look at the code and think it was an oversight that hidden commits could affect the prefix length.

Reply:
Ok I updated the first paragraph of the trait documentation as follows. This is even less ambiguous:

```
    /// This function returns the shortest length of a prefix of `key` that
    /// disambiguates it from every other key in the index, including hidden
    /// keys. Since it disambiguates among all keys, the returned length is
    /// usually a few digits longer than the minimum length necessary to
    /// disambiguate among the visible keys only.
```

How's that?

Reply:
Resolving since we are discussing in a different thread.